receiver/otlpreceiver/trace/otlp.go

```diff
@@ -72,9 +72,16 @@ func (r *Receiver) Export(ctx context.Context, req *collectortrace.ExportTraceSe
 	for _, rss := range req.ResourceSpans {
 		for _, ils := range rss.InstrumentationLibrarySpans {
 			for _, span := range ils.Spans {
-				if span.Status.Code == otlptrace.Status_STATUS_CODE_UNSET &&
-					span.Status.DeprecatedCode != otlptrace.Status_DEPRECATED_STATUS_CODE_OK {
-					span.Status.Code = otlptrace.Status_STATUS_CODE_ERROR
+				switch span.Status.Code {
+				case otlptrace.Status_STATUS_CODE_UNSET:
+					if span.Status.DeprecatedCode != otlptrace.Status_DEPRECATED_STATUS_CODE_OK {
+						span.Status.Code = otlptrace.Status_STATUS_CODE_ERROR
+					}
+				case otlptrace.Status_STATUS_CODE_OK:
+					// If status code is set then overwrites deprecated.
+					span.Status.DeprecatedCode = otlptrace.Status_DEPRECATED_STATUS_CODE_OK
+				case otlptrace.Status_STATUS_CODE_ERROR:
+					span.Status.DeprecatedCode = otlptrace.Status_DEPRECATED_STATUS_CODE_UNKNOWN_ERROR
```

Comment:
Why these changes? The spec says that receivers MUST:

> //   If code!=STATUS_CODE_UNSET then the value of `deprecated_code` MUST be
>   //   ignored, the `code` field is the sole carrier of the status.

This does not seem to match the spec anymore.

Reply:
What does the spec says? Why does "SetCode" updates deprecated but in this case we don't?

Reply:
Updated comments to SetCode as well as for the receiver to clarify when we are sender vs receiver.

Reply:
Discussed offline, so yes, OTLP receiver is both a "receiver" and "sender" in the sense that the proto uses these words here: https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L239